### DISCUSSION TITLE
Support of ring_hash for consistent balancing

### DISCUSSION THREAD

**Discussion by BorshchevY** [envoyproxy/gateway#4609] [Q&A] [answered]
Hi guys!
We use Envoy Proxy as the core router in our Kubernetes cluster now. We also use the ring_hash algorithm for load balancing the traffic for some of our backends. We are planning to migrate to Envoy Gateway, but at the moment, as I see it, this concept doesn't support the ring_hash algorithm for consistent lb. I wanted to clarify why there is no support for this algorithm in the current versions and whether support is planned for future releases?

**Comment by arkodg** [comment, marked as answer]
hey we currently support/implement Maglev for the consistent hashing load balancing type
https://gateway.envoyproxy.io/docs/tasks/traffic/load-balancing/#consistent-hash

https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/upstream/load_balancing/load_balancers#maglev highlights that this can be used as a drop in replacement for ring hash and performs better

**Reply by BorshchevY** [reply]
Got it. Thanks for clarifying that.
